packaging/cross_build.sh

```diff
@@ -0,0 +1,99 @@
+#!/bin/bash
+#
+# cross_build.sh
+#
+# Build rust project for various targets
+#
+# Meant for use during the final release as part of CI, but can be used for
+# local testing too
+#
+# Usage: cross_build.sh TARGET
+#   For supported TARGET values, see Cross.toml in project root.
+#
+# Example: cross_build.sh x86_64-unknown-linux-gnu
+
+# exit the script when a command fails
+set -o errexit
+
+# catch exit status for piped commands
+set -o pipefail
+
+TARGET=$1
+if [ -z "$TARGET" ]; then
+  echo "Usage: ${0##*/} TARGET"
+  exit 1
+fi
+
+if [ "$TREMOR_MODE" = "debug" ]; then
+  BUILD_MODE=debug
+else
+  BUILD_MODE=release
+fi
+
+BIN_NAME="tremor-server"
+
+ROOT_DIR="$(git rev-parse --show-toplevel)"
+
+echo "Building ${BIN_NAME} for target ${TARGET} in ${BUILD_MODE} mode, from ${ROOT_DIR}"
+
+# this move allows us to run this script from anywhere in the repo
+pushd "$ROOT_DIR" > /dev/null
+
+###############################################################################
+
+# install cross if not already there (helps us build easily across various targets)
+# see https://github.com/rust-embedded/cross
+#
+# currently need to install it from a personal fork for builds to work against custom targets
+# (eg: x86_64-alpine-linux-musl which we use for generating working musl binaries right now)
+if ! command -v cross > /dev/null 2>&1; then
+  echo "Installing cross..."
+  cargo install --git https://github.com/anupdhml/cross.git --branch custom_target_fixes
```

Comment:
Can we upstream that?

Reply:
will do. I wanted to try out some other things on that branch before filing a PR -- will test them out soon and submit it upstream.

Reply:
got some time today to finally make that upstream PR!

https://github.com/rust-embedded/cross/pull/431